v2/cmd/gen/gen.go

```diff
@@ -0,0 +1,234 @@
+package gen
+
+import (
+	"fmt"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/cosmos/iavl-bench/bench"
+	"github.com/cosmos/iavl/v2"
+	"github.com/cosmos/iavl/v2/testutil"
+	"github.com/dustin/go-humanize"
+	"github.com/kocubinski/costor-api/compact"
+	"github.com/kocubinski/costor-api/core"
+	"github.com/rs/zerolog"
+	zlog "github.com/rs/zerolog/log"
+	"github.com/spf13/cobra"
+)
+
+var log = zlog.Output(zerolog.ConsoleWriter{
+	Out:        os.Stderr,
+	TimeFormat: time.Stamp,
+})
+
+func Command() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "gen",
+		Short: "generate changesets",
+	}
+
+	cmd.AddCommand(emitCommand(), treeCommand())
+
+	return cmd
+}
+
+func getChangesetIterator(typ string) (bench.ChangesetIterator, error) {
+	switch typ {
+	case "osmo-like":
+		return testutil.OsmoLike().Iterator, nil
+	case "osmo-like-many":
+		return testutil.OsmoLikeManyTrees().Iterator, nil
+	case "height-zero":
+		return testutil.NewTreeBuildOptions().Iterator, nil
+	default:
+		return nil, fmt.Errorf("unknown generator type %s", typ)
+	}
+}
+
+func emitCommand() *cobra.Command {
+	var (
+		typ   string
+		out   string
+		start int
+		limit int
+	)
+	cmd := &cobra.Command{
+		Use:   "emit",
+		Short: "emit generated changesets to disk",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			itr, err := getChangesetIterator(typ)
+			if err != nil {
+				return err
+			}
+			ctx := core.Context{Context: cmd.Context()}
+
+			stream := compact.StreamingContext{
+				In:          make(chan compact.Sequenced),
+				Context:     ctx,
+				OutDir:      out,
+				MaxFileSize: 100 * 1024 * 1024,
+			}
+
+			var wg sync.WaitGroup
+			wg.Add(1)
+			go func() {
+				stats, err := stream.Compact()
+				if err != nil {
+					log.Fatal().Err(err).Msg("failed to compact")
```

Comment:
It's not recommended to invoke log.Fatal in a go routine, it doesn't give others time to cleanup.